### DISCUSSION TITLE
Could not Find liblibrustdesk.so error when running with flutter

### DISCUSSION THREAD

**Discussion by Ahsan161815** [rustdesk/rustdesk#521] [General]
I am trying to run RustDesk android https://github.com/rustdesk/rustdesk/tree/master/flutter with flutter-sdk 2.10.0 and getting error [liblibrustdesk.so](https://liblibrustdesk.so/) not Found.

I am following these instruction for flutter build https://rustdesk.com/docs/en/dev/build/android/

Thanks

**Comment by rustdesk** [comment]
Friendly reminder 1: **`Staring`** the GitHub repo is polite and friendly before asking questions, and the person being asked will be happier to answer your questions.

Friendly reminder 2: Please provide more details, what errors have you seen while following our instructions. Make sure you run on Linux.

**Reply by sany1231** [reply]
Hi, as this documentation says, we shloud download so.tar.gz, extract and copy liblibrustesk.so but this file is not part of the folder
https://rustdesk.com/docs/en/dev/build/android/

**Comment by sany1231** [comment]
Hi, i have same problme, i follow android build guide https://rustdesk.com/docs/en/dev/build/android/

when i try to copy
cp ../target/aarch64-linux-android/release/liblibrustdesk.so android/app/src/main/jniLibs/arm64-v8a/librustdesk.so

there is no liblibrustdesk.so only liblibrustdesk.d or liblibrustdesk.rlib

Should i run something to generate this file or it is simply missing from repo ?

Thanks.

**Comment by Orholt** [comment]
So, you need to build this file.
Remember that official build instructions are not done in 100% so they are not cover everything.

If you really want to be sure that you're done everything correctly consider following [worklfow file](https://github.com/rustdesk/rustdesk/blob/master/.github/workflows/flutter-ci.yml) (Mostly generate-bridge and build-apk part)

**Reply by sany1231** [reply]
Problem is that work flows are only for arm64 not for armv7, i am able to build librustesk.so for arm64 but for armv7 no luck.

**Reply by Orholt** [reply]
Hmm, what is your rust target architecture?

**Reply by sany1231** [reply]
cargo ndk --platform 21 --target armv7-linux-androideabi build --release --features flutter

**Reply by Orholt** [reply]
To be honest, I could try to build that on my machine but if even workflow does not work idk how to help you.

**Reply by sany1231** [reply]
there is no workflow for armv7a :(
but thanks for ur time

**Reply by sany1231** [reply]
but if u have some spare time u can try :)
